mne_sandbox/connectivity/cfc.py

```diff
@@ -0,0 +1,506 @@
+    pac_out : array, dtype float, shape (n_pairs, [n_events])
+        The computed phase-amplitude coupling between each pair of data sources
+        given in ixs.
+    """
+    from ..externals.pacpy import pac as ppac
+    if pac_func not in _pac_funcs:
+        raise ValueError("PAC function {0} is not supported".format(pac_func))
+    func = getattr(ppac, pac_func)
+    ixs = np.array(ixs, ndmin=2)
+
+    if data.ndim != 2:
+        raise ValueError('Data must be shape (n_channels, n_times)')
+    if ixs.shape[1] != 2:
+        raise ValueError('Indices must have have a 2nd dimension of length 2')
+    if len(f_phase) != 2 or len(f_amp) != 2:
+        raise ValueError('Frequencies must be specified w/ a low/hi tuple')
+
+    print('Pre-filtering data and extracting phase/amplitude...')
+    hi_phase = pac_func in _hi_phase_funcs
+    data_ph, data_am, ix_map_ph, ix_map_am = _pre_filter_ph_am(
+        data, sfreq, ixs, f_phase, f_amp, npad=npad, hi_phase=hi_phase,
+        scale_amp_func=scale_amp_func)
+    ixs_new = [(ix_map_ph[i], ix_map_am[j]) for i, j in ixs]
+
+    if ev is not None:
+        use_times = [tmin, tmax] if use_times is None else use_times
+        ev_grouping = np.ones_like(ev) if ev_grouping is None else ev_grouping
+        data_ph, times, msk_ev = _array_raw_to_epochs(
+            data_ph, sfreq, ev, tmin, tmax)
+        data_am, times, msk_ev = _array_raw_to_epochs(
+            data_am, sfreq, ev, tmin, tmax)
+
+        # In case we cut off any events
+        ev, ev_grouping = [i[msk_ev] for i in [ev, ev_grouping]]
+
+        # Baselining before returning
+        rescale(data_am, times, baseline, baseline_kind, copy=False)
+        msk_time = _time_mask(times, *use_times)
+        data_am, data_ph = [i[..., msk_time] for i in [data_am, data_ph]]
+
+        # Stack epochs to a single trace if specified
+        if concat_epochs is True:
+            ev_unique = np.unique(ev_grouping)
+            concat_data = []
+            for i_ev in ev_unique:
+                msk_events = ev_grouping == i_ev
+                concat_data.append([np.hstack(i[msk_events])
+                                    for i in [data_am, data_ph]])
+            data_am, data_ph = zip(*concat_data)
+    else:
+        data_ph, data_am = [data_ph], [data_am]
+
+    n_ep = len(data_ph)
+    pac = np.zeros([n_ep, len(ixs_new)])
+    pbar = ProgressBar(n_ep)
+    for iep, (ep_ph, ep_am) in enumerate(zip(data_ph, data_am)):
+        for iix, (i_ix_ph, i_ix_am) in enumerate(ixs_new):
+            # f_phase and f_amp won't be used in this case
+            pac[iep, iix] = func(ep_ph[i_ix_ph], ep_am[i_ix_am],
+                                 f_phase, f_amp, filterfn=False)
+        pbar.update_with_increment_value(1)
+    if return_data:
+        return pac, data_ph, data_am
+    else:
+        return pac
+
+
+def _pre_filter_ph_am(data, sfreq, ixs, f_ph, f_am, filterfn=None,
+                      npad=None, hi_phase=False, scale_amp_func=None,
```

Comment:
I can change this to 'auto' - I think I went with None originally because that's what the current release of MNE uses, but I think in the next release it will switch to 'auto', so should probably just change it